python/dgl/dataloading/pytorch/__init__.py

```diff
@@ -242,9 +297,25 @@ def __init__(self, g, nids, block_sampler, device='cpu', **kwargs):
             self.is_distributed = True
         else:
             self.collator = _NodeCollator(g, nids, block_sampler, **collator_kwargs)
-            self.dataloader = DataLoader(self.collator.dataset,
-                                         collate_fn=self.collator.collate,
-                                         **dataloader_kwargs)
+            if "num_workers" in dataloader_kwargs and dataloader_kwargs["num_workers"] > 0:
+                assert g.device == th.device('cpu'), \
+                    'Only graphs on the cpu are supported for NodeDataLoader ' \
+                    'in the case of num_workers > 0.'
+                self.dataloader = DataLoader(
+                    self.collator.dataset,
+                    collate_fn=self.collator.collate,
+                    **dataloader_kwargs)
+            else:
+                if isinstance(self.collator.dataset, th.Tensor):
+                    self.dataloader = _ScalarDataLoader(
+                        self.collator.dataset,
+                        collate_fn=self.collator.collate,
+                        **dataloader_kwargs)
```

Comment:
there might be compatibility issues. not all pytorch DataLoader's arguments are supported here.

Reply:
I changed it to the pytorch prescribed method of implementing an IterableDataset, and using that to control the batches in 
89d8fce